File: quickstart/java/src/test/java/TalonTest.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

import edu.wpi.first.wpilibj.RobotBase;
import edu.wpi.first.wpilibj.Talon;
import edu.wpi.first.wpilibj.sim.PWMSim;

public class TalonTest {
  @Test
  public void TestTalonSet() {
    RobotBase.initializeHardwareConfiguration();
    try (Talon talon = new Talon(3)) {
      PWMSim talonSim = new PWMSim(3);
      talon.set(0.5);
      assertEquals(0.5, talonSim.getSpeed(), 0.001);
      talon.set(-0.5);
      assertEquals(-0.5, talonSim.getSpeed(), 0.001);
    }
  }
}

File: quickstart/cpp/src/test/TalonTest.cpp
#include "gtest/gtest.h"
#include "HAL/HAL.h"
#include "Simulation/PWMSim.h"
#include "Talon.h"

TEST(TalonTests, TestTalonSet) {
  frc::Talon talon{3};
  frc::sim::PWMSim talonSim{3};
  talon.Set(0.5);
  ASSERT_DOUBLE_EQ(0.5, talonSim.GetSpeed());
  talon.Set(-0.5);
  ASSERT_DOUBLE_EQ(-0.5, talonSim.GetSpeed());
}


